homeassistant/components/zeroconf/__init__.py

```diff
@@ -557,7 +557,9 @@ def info_from_service(service: AsyncServiceInfo) -> ZeroconfServiceInfo | None:
     )
 
 
-def _first_non_link_local_or_v6_address(addresses: list[bytes]) -> str | None:
+def _first_non_link_local_or_v6_address(
```

Comment:
We should probably change this to prefer ipv4 addresses since that what is actually did before given `addresses` could never have ipv6

Reply:
Also we need to label this as a breaking change since `host` will now have ipv6 addresses if ipv4 is not available which will cause something to break.  I don't think there is a way to avoid that though

Reply:
Okay. The blast radius should be fairly small here as I don't think there are many IPv6-only devices out there.

Reply:
We need to update the breaking change text to tell developers to use the new utils from https://github.com/home-assistant/core/pull/66472

Reply:
We should assume most integrations developers don't understand how zeroconf works and may have never worked with IPv6

Reply:
Updated